File: 2015/05/1.go
package main

import (
	"fmt"
	"os"
	"regexp"
	"strings"
)

func main() {
	input, err := os.Open("./2015/05/input.txt")
	if err != nil {
		fmt.Fprintln(os.Stderr, "Unable to open input file")
		os.Exit(1)
	}
	stat, _ := input.Stat()
	b := make([]byte, stat.Size())
	input.Read(b)
	niceCount := 0
	for _, line := range strings.Split(string(b), "\n") {
		vowelsCount := 0
		// We can't use "(.){2}" a regeex as Go's regex is broken
		atleastOneLetterThatAppearTwiceInARow := false
		cold := "this is a joke string"
		for _, c := range strings.Split(line, "") {
			if c == cold {
				atleastOneLetterThatAppearTwiceInARow = true
			}
			switch c {
			case "a", "A", "e", "E", "i", "I", "o", "O", "u", "U":
				vowelsCount++
			}
			cold = c
		}
		hasAtleastThreeVowels := vowelsCount >= 3
		hasForbiddenStrings := regexp.MustCompile("ab|cd|pq|xy").MatchString(line)
		if hasAtleastThreeVowels && atleastOneLetterThatAppearTwiceInARow && !hasForbiddenStrings {
			niceCount++
		}
	}
	fmt.Println(niceCount)
}

File: 2015/05/1.js
"use strict";
const fs = require("node:fs");
const crypto = require("node:crypto");
let contents = fs.readFileSync("./2015/05/input.txt", "utf8").toString();

let niceCount = 0;
for (const line of contents.split("\n")) {
  if (
    /(.)\1/.test(line) &&
    /([aeiouAEIOU]).*?([aeiouAEIOU]).*?([aeiouAEIOU])/.test(line) &&
    !/ab|cd|pq|xy/.test(line)
  ) {
    niceCount++;
  }
}
console.log(niceCount);
